packages/@expo/use-updates/src/UseUpdates.ts

```diff
@@ -0,0 +1,134 @@
+import * as Updates from 'expo-updates';
+import { useEffect, useRef, useState } from 'react';
+
+import type {
+  UseUpdatesCallbacksType,
+  UseUpdatesReturnType,
+  UseUpdatesStateType,
+} from './UseUpdates.types';
+import {
+  checkForUpdateAndReturnAvailableAsync,
+  currentlyRunning,
+  downloadUpdateAsync,
+  runUpdateAsync,
+  availableUpdateFromEvent,
+} from './UseUpdatesUtils';
+
+/**
+ * Hook that obtains the Updates info structure and functions.
+ *
+ * @param callbacks Optional set of callbacks that will be called when `checkForUpdate()`, `downloadUpdate()`, `downloadAndRunUpdate()`, or `runUpdate()`, start, complete, or have errors.
+ *
+ * @return the [`UpdatesInfo`](#updatesinfo) structure and associated methods. When using this hook, the methods returned should be used instead of [`checkForUpdateAsync`](#updatescheckforupdateasync), [`fetchUpdateAsync`](#updatesfetchupdateasync), [`readLogEntriesAsync`](#updatesreadlogentriesasync), and [`reloadAsync`](#updatesreloadasync).
+ *
+ * @example
+ * ```jsx UpdatesDemo.tsx
```

Comment:
```suggestion
 * ```tsx UpdatesDemo.tsx
```

JSDoc nit.

Reply:
I'm not sure I understand the reason behind why this example is using TypeScript code highlighting or using the file format. I don't see any TS code being used below. For consistency I'd keep it JS or JSX if there is no TS Code.

Reply:
Nice catch, then we should change the file extension to the `jsx` I guess.

```suggestion
 * ```jsx UpdatesDemo.jsx
```